### PR TITLE
Added config default values

### PR DESCRIPTION
**Pull request by breakreturn**
Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,12 +118,62 @@ if(ENABLE_ALL_DOC)
...
+
+#
 # RPATH options
 #
 
 # Use RPATH only if we're not installing to a system directory
-list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_FULL_LIBDIR}" IS_SYSTEM_DIR)
-if(${IS_SYSTEM_DIR} STREQUAL "-1")
+ list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_FULL_LIBDIR}" IS_SYSTEM_DIR)
+ if(${IS_SYSTEM_DIR} STREQUAL "-1")
```
Comment (DunderRoffe) [inline CMakeLists.txt]: why this space?

File: agent/softwarecontainer-config.in [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
+# shutdown-timeout = @SHUTDOWN_TIMEOUT@
 
 # Location for all sharted host/container mounts
-shared-mounts-dir = @SHARED_MOUNTS_DIR@
+# shared-mounts-dir = @SHARED_MOUNTS_DIR@
 
 # Path to the LXC configuration used with liblxc
 # NOTE: This option is being deprecated
-deprecated-lxc-config-path = @LXC_CONFIG_PATH@
+# deprecated-lxc-config-path = @LXC_CONFIG_PATH@
```
Comment (DunderRoffe) [inline agent/softwarecontainer-config.in:31]: Should we really keep deprecated stuff?

File: agent/src/main.cpp [hunk 3]
```diff
@@ -137,8 +138,8 @@ int main(int argc, char **argv)
     int userID = 0;
     bool keepContainersAlive = false;
     int timeout = -2;
-    Glib::ustring serviceManifestDir = SERVICE_MANIFEST_DIR_OPT;
-    Glib::ustring defaultServiceManifestDir = DEFAULT_SERVICE_MANIFEST_DIR_OPT;
+    Glib::ustring serviceManifestDir = "";
+    Glib::ustring defaultServiceManifestDir = "";
```
Comment (DunderRoffe) [inline agent/src/main.cpp]: Would be nice to have a constant that these are set to that can be used to check if they were set later.
Remove magic values.

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

File: agent/softwarecontainer-config.in (hunk 2 above)
Comment (breakreturn) [reply agent/softwarecontainer-config.in:31]: No we should remove it. We have an action to do that planned in the backlog. But we need it at the moment so it has to be there. My thought was that this name will make sure the user expects this to change, and it's easier to find as it sticks out. Do you think it's OK to keep this as it is until we actually remove the need?

Comment (DunderRoffe) [reply agent/softwarecontainer-config.in:31]: Yes, I just wanted to make sure that this was according to plan. :)

File: agent/softwarecontainer-config.in [hunk 4]
```diff
@@ -1,27 +1,35 @@
+
+#
+# General SoftwareContainer configuration.
+#
+# Optional values are commented out. All other values are mandatory.
```
Comment (tobsan) [inline agent/softwarecontainer-config.in]: Can we add that the optional values we have commented out have the default values in this file.

File: agent/src/main.cpp (hunk 3 above)
Comment (tobsan) [reply agent/src/main.cpp]: With the config parser we're using, that is not currently possible, unless we write our own callbacks (which feels a bit overkill imo)

Comment (breakreturn) [reply agent/src/main.cpp]: I agree. It's fixed with the next update.

**Review by tobsan — CHANGES_REQUESTED** [review]

**Review by DunderRoffe — APPROVED** [review]

File: agent/softwarecontainer-config.in (hunk 4 above)
Comment (breakreturn) [reply agent/softwarecontainer-config.in]: Like: "Optional configs are commented out, but are set to the default values that will be used by SoftwareContainer if nothing else is specified." ?

**Review by tobsan — APPROVED** [review]

**Pull request merged**